bpf/flows.c

```diff
@@ -325,6 +336,66 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
     }
     return TC_ACT_OK;
 }
+static inline int export_packet_payload (struct __sk_buff *skb) {
+    void *data_end = (void *)(long)skb->data_end;
+    void *data = (void *)(long)skb->data;
+    payload_meta meta;
+    //packet meta;
+    struct ethhdr *eth  = data;
+    struct iphdr  *ip;
+    struct udphdr *udp_data;
+    
+    
+    
+    if ((void *)eth + sizeof(*eth) > data_end) {
+       return TC_ACT_UNSPEC;	
+    }
+
+    ip = data + sizeof(*eth);
+    if ((void *)ip + sizeof(*ip) > data_end) {
+       return TC_ACT_UNSPEC;	
+    }
+
+    udp_data = (void *)ip + sizeof(*ip);
+    if ((void *)udp_data + sizeof(*udp_data) > data_end) {
+       return TC_ACT_UNSPEC;	
+    }
+
+    if (eth->h_proto != __bpf_htons(ETH_P_IP)) {
+       return TC_ACT_UNSPEC;	
+    }
+
+    //Only analyze UDP packets
+    if (ip->protocol != IPPROTO_UDP ) {
+       return TC_ACT_UNSPEC;	
+    }
+
+    __be16 port = udp_data->dest;
+    //TODO: Update port number/filters to be read from ENV variable
+    __be16 portFromPanoFilter = __bpf_htons(53);
```

Comment:
is the oplan to use only one port as filter or we can also provide range ?

Reply:
Eventually we would want this filter to be able to serve a number of possible subsets of packets. 
Ad of now, we only want to use it for port 53 for DNS.

Reply:
Can you make this a macro, hardcode of the port inside the code is less readable.
On that front if comparing ports then enter reverse endian format so that you save on on `bpf_htons` while comparing.

Also might be nitpicking but can you rename the variable from `portFromPanoFilter` to `filter_port` or something else, C naming convention does not like camel case.